pkg/segment/aggregations/streamstats.go

```diff
@@ -62,12 +65,37 @@ func InitRangeStat() *structs.RangeStat {
 	}
 }
 
-func PerformNoWindowStreamStatsOnSingleFunc(ssOption *structs.StreamStatsOptions, ssResults *structs.RunningStreamStatsResults, measureAgg utils.AggregateFunctions, colValue interface{}) (float64, bool, error) {
-	result := ssResults.CurrResult
+func getValuesNoWindow(valuesMap map[string]struct{}) string {
+	uniqueStrings := make([]string, 0)
+	for str := range valuesMap {
+		uniqueStrings = append(uniqueStrings, str)
+	}
+	sort.Strings(uniqueStrings)
+	return strings.Join(uniqueStrings, "&nbsp")
+}
+
+func getValuesWindow(valuesMap map[string]int) string {
+	uniqueStrings := make([]string, 0)
+	for str := range valuesMap {
+		uniqueStrings = append(uniqueStrings, str)
+	}
+	sort.Strings(uniqueStrings)
+	return strings.Join(uniqueStrings, "&nbsp")
+}
```

Comment:
addressed in the latest commit.

Reply:
Keeping two maps is useful for noWindow because map[string]struct{} uses less memory compared to map[string]int.
For the case of noWindow, we just need the uniqueStrings and memory efficiency since the unique values can be quite large.